nixos/modules/services/web-apps/open-web-calendar.nix

```diff
@@ -0,0 +1,166 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  inherit (lib)
+    mkIf
+    mkOption
+    mkEnableOption
+    mkPackageOption
+    mkDefault
+    types
+    concatMapStringsSep
+    generators
+    ;
+  cfg = config.services.open-web-calendar;
+
+  finalPackage = cfg.package.overrideAttrs (old: {
+    # The calendarSettings need to be merged with the default_specification.yml
+    # in the source. This way we use upstreams default values but keep everything overridable.
+    prePatch =
+      let
+        nixosSpec = calendarSettingsFormat.generate "nixos_specification.json" cfg.calendarSettings;
+      in
+      ''
+        ${pkgs.yq}/bin/yq -s '.[0] * .[1]' open_web_calendar/default_specification.yml ${nixosSpec} > newspec.json
+        mv newspec.json open_web_calendar/default_specification.yml
+      '';
+  });
```

Comment:
I think this could be solved using `override` instead.

One way to do it would be this:

```diff
diff --git a/nixos/modules/services/web-apps/open-web-calendar.nix b/nixos/modules/services/web-apps/open-web-calendar.nix
index 70db9e1fa216..863e8881a258 100644
--- a/nixos/modules/services/web-apps/open-web-calendar.nix
+++ b/nixos/modules/services/web-apps/open-web-calendar.nix
@@ -18,18 +18,15 @@ let
     ;
   cfg = config.services.open-web-calendar;
 
-  finalPackage = cfg.package.overrideAttrs (old: {
+  nixosSpec = calendarSettingsFormat.generate "nixos_specification.json" cfg.calendarSettings;
+  finalPackage = cfg.package.override {
     # The calendarSettings need to be merged with the default_specification.yml
     # in the source. This way we use upstreams default values but keep everything overridable.
-    prePatch =
-      let
-        nixosSpec = calendarSettingsFormat.generate "nixos_specification.json" cfg.calendarSettings;
-      in
-      ''
-        ${pkgs.yq}/bin/yq -s '.[0] * .[1]' open_web_calendar/default_specification.yml ${nixosSpec} > newspec.json
-        mv newspec.json open_web_calendar/default_specification.yml
-      '';
-  });
+    defaultSpecificationFile = pkgs.runCommand "custom-default_specification.yml" {} ''
+      ${pkgs.yq}/bin/yq -s '.[0] * .[1]' ${cfg.package}/${cfg.package.defaultSpecificationPath} ${nixosSpec} > $out
+    '';
+  };
+
   inherit (finalPackage) python;
   pythonEnv = python.buildEnv.override { extraLibs = [ (python.pkgs.toPythonModule finalPackage) ]; };
 
diff --git a/pkgs/by-name/op/open-web-calendar/package.nix b/pkgs/by-name/op/open-web-calendar/package.nix
index a973d0c6aa73..1d1f0ace7145 100644
--- a/pkgs/by-name/op/open-web-calendar/package.nix
+++ b/pkgs/by-name/op/open-web-calendar/package.nix
@@ -3,6 +3,8 @@
   python3,
   fetchPypi,
   nixosTests,
+
+  defaultSpecificationFile ? null,
 }:
 
 let
@@ -26,6 +28,10 @@ python.pkgs.buildPythonApplication rec {
     ln -s $PWD/features open_web_calendar/features
   '';
 
+  postInstall = lib.optionalString (defaultSpecificationFile != null) ''
+    install -D ${defaultSpecificationFile} $out/$defaultSpecificationPath
+  '';
+
   build-system = with python.pkgs; [
     hatchling
     hatch-vcs
@@ -55,6 +61,8 @@ python.pkgs.buildPythonApplication rec {
 
   pythonImportsCheck = [ "open_web_calendar.app" ];
 
+  defaultSpecificationPath = "${python.sitePackages}/open_web_calendar/default_specification.yml";
+
   passthru = {
     inherit python;
     tests = {

```